types/jasmine/index.d.ts

```diff
@@ -247,11 +248,11 @@ declare namespace jasmine {
     function objectContaining<T>(sample: Partial<T>): ObjectContaining<T>;
     function createSpy(name?: string, originalFn?: Function): Spy;
 
-    function createSpyObj(baseName: string, methodNames: SpyObjMethodNames): any;
-    function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames<T>): SpyObj<T>;
+    function createSpyObj(baseName: string, methodNames: SpyObjMethodNames, propertyNames?: SpyObjMethodNames): any;
+    function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames<T>, propertyNames?: SpyObjMethodNames<T>): SpyObj<T>;
 
-    function createSpyObj(methodNames: SpyObjMethodNames): any;
-    function createSpyObj<T>(methodNames: SpyObjMethodNames<T>): SpyObj<T>;
+    function createSpyObj(methodNames: SpyObjMethodNames, propertyNames?: SpyObjMethodNames): any;
```

Comment:
Instead of making it optional, I'd suggest to create a new overload and mark it with `@since 3.5.0` attribute. E.g.:

```ts
    /**
     * @since 3.5.0
     */
    function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames<T>, propertyNames: SpyObjPropertyNames<T>): SpyObj<T>;
```

Do the same for all the methods.